cmd/metric-adapter/main.go

```diff
@@ -50,19 +49,13 @@ type MetricAdapter struct {
 	Message string
 }
 
-func (a *MetricAdapter) makeProvider(remoteAdapter *metricprovider.RemoteAdapter, config *rest.Config, client client.Client) (provider.CustomMetricsProvider, error) {
-	kubeClient, err := kubernetes.NewForConfig(config)
-	if err != nil {
-		return nil, fmt.Errorf("unable to create kube client: %v", err)
-	}
+func (a *MetricAdapter) makeProvider(remoteAdapter *metricprovider.RemoteAdapter, client client.Client, recorder record.EventRecorder) provider.CustomMetricsProvider {
```

Comment:
is initProvider better than makeProvider?
same comment to makeExternalMetricProvider

Reply:
> is initProvider better than makeProvider? same comment to makeExternalMetricProvider

why? What is the difference?

Reply:
just a minor comment, ok for either way, initProvider is what is used in kubernetes cloud provider